### PR TITLE
docs(faq): using shorebird for all dart changes

### PR DESCRIPTION
**Pull request by felangel**
## Status

**READY**

## Description

- docs(faq): using shorebird for all dart changes

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -111,6 +111,16 @@ Also Shorebird does not support changing native code (e.g. Java/Kotlin on
...
 attempted patch if you have changed native code.
 
+### Can I use Shorebird for all my Dart changes?
+
+Technically speaking, yes, a Shorebird patch can update any Dart code.
+Note that depending on how you distribute your apps, some store agreements
+expect feature changes to go through store review.
+Notably Apple's App Store requires that an update "does not change the
+primary purpose of the Application by providing features or functionality
+that are inconsistent with the intended and advertised purpose of the Application.
```
Comment (bryanoltman) [inline docs/faq.md:121]:
```suggestion
that are inconsistent with the intended and advertised purpose of the Application."
```

Comment (felangel) [reply docs/faq.md]: Good catch, thanks!

File: docs/faq.md [hunk 2]
```diff
@@ -111,6 +111,16 @@ Also Shorebird does not support changing native code (e.g. Java/Kotlin on
...
+Also note that more Dart code changes will result in larger patch sizes.
```
Comment (eseidel) [inline docs/faq.md:122]:
```suggestion
Also note that patch sizes correlate with the total amount of Dart changed from the original released app (each patch is a diff against the released dart code, not a diff to the previous patch).
```

File: docs/faq.md [hunk 3]
```diff
@@ -111,6 +111,16 @@ Also Shorebird does not support changing native code (e.g. Java/Kotlin on
 Android or Objective-C/Swift on iOS). The tool will warn you during an
 attempted patch if you have changed native code.
 
+### Can I use Shorebird for all my Dart changes?
+
+Technically speaking, yes, a Shorebird patch can update any Dart code.
```
Comment (bryanoltman) [inline docs/faq.md:116]:
```suggestion
Shorebird can be used to update any and all Dart code in your app, including non-plugin dependencies.
```

File: docs/faq.md [hunk 4]
```diff
@@ -111,6 +111,16 @@ Also Shorebird does not support changing native code (e.g. Java/Kotlin on
...
+Note that depending on how you distribute your apps, some store agreements
```
Comment (bryanoltman) [inline docs/faq.md:117]:
```suggestion
However, some store agreements
```

**Review by eseidel — APPROVED** [review]

File: docs/faq.md (hunk 3 above)
Comment (felangel) [reply docs/faq.md:116]: Maybe we should say "include dart packages (not plugins)."

Comment (bryanoltman) [reply docs/faq.md:116]: "including packages that do not contain plugins"?

File: docs/faq.md (hunk 4 above)
Comment (felangel) [reply docs/faq.md:117]: I think leaving it as is is more explicit personally 🤷

File: docs/faq.md (hunk 3 above)
Comment (felangel) [reply docs/faq.md]: I went with "including pure dart packages" lmk what you think

**Pull request merged**